framework/contracts/account/src/sub_account.rs

```diff
@@ -184,3 +185,56 @@ pub fn maybe_update_sub_account_governance(deps: DepsMut) -> AccountResult<Vec<C
 
     Ok(msgs)
 }
+
+/// Renounce ownership of this account \
+/// **WARNING**: This will lock the account, making it unusable.
+pub fn remove_account_from_contracts(deps: DepsMut) -> AccountResult<Vec<CosmosMsg>> {
+    let mut msgs = vec![];
+
+    let account_id = ACCOUNT_ID.load(deps.storage)?;
+    // Check for any sub accounts
+    let sub_account = SUB_ACCOUNTS
+        .keys(deps.storage, None, None, cosmwasm_std::Order::Ascending)
+        .next()
+        .transpose()?;
+    ensure!(
+        sub_account.is_none(),
+        AccountError::RenounceWithSubAccount {}
+    );
+
+    let ownership = ownership::get_ownership(deps.storage)?;
+    if let GovernanceDetails::SubAccount { account } = ownership.owner {
+        // Unregister itself (sub-account) from the owning account.
+        msgs.push(
+            wasm_execute(
+                account,
+                &ExecuteMsg::UpdateSubAccount(UpdateSubAccountAction::UnregisterSubAccount {
+                    id: account_id.seq(),
+                }),
+                vec![],
+            )?
+            .into(),
+        );
+    }
+
+    let config = CONFIG.load(deps.storage)?;
+    let vc = VersionControlContract::new(config.version_control_address);
+    let mut namespaces = vc
+        .query_namespaces(vec![account_id], &deps.querier)?
+        .namespaces;
+    let namespace = namespaces.pop();
+    if let Some((namespace, _)) = namespace {
+        // Remove the namespace that this account holds.
+        msgs.push(
+            wasm_execute(
+                vc.address,
+                &abstract_std::version_control::ExecuteMsg::RemoveNamespaces {
+                    namespaces: vec![namespace.to_string()],
+                },
+                vec![],
+            )?
+            .into(),
```

Comment:
This namespace can be used by other users after that right? Remove confusing name for it. And do we still want it to use array of namespaces?

Reply:
It's nice for us to be able to batch removals if we want. 

Which name would you suggest? "ForgoNamespace"?

Reply:
Yeah or return/disclaim

Reply:
@Kayanski what do you think?

Reply:
Let's do it in a different PR: https://linear.app/abstract-sdk/issue/ABS-538/rename-removenamespace-on-vc